File: frontend/src/api/ai.ts
import api from './client';

export interface AILesson {
  title: string;
  description: string;
  lesson_type: string;
  order: number;
  duration_seconds: number;
}

export interface AIModule {
  title: string;
  description: string;
  order: number;
  lessons: AILesson[];
}

export interface AIGeneratedCourse {
  title: string;
  slug: string;
  short_description: string;
  description: string;
  level: string;
  category: string;
  modules: AIModule[];
}

export async function generateCourse(prompt: string): Promise<AIGeneratedCourse> {
  return api.request<AIGeneratedCourse>('/ai/generate-course', {
    method: 'POST',
    body: JSON.stringify({ prompt }),
  });
}

// ── Quiz generation ──────────────────────────────────────────────────────

export interface AIQuizOption {
  label: string;
  text: string;
  is_correct: boolean;
}

export interface AIQuizQuestion {
  text: string;
  order: number;
  question_type: string;
  explanation: string;
  points: number;
  options: AIQuizOption[];
}

export interface AIGeneratedQuiz {
  title: string;
  passing_score_percent: number;
  questions: AIQuizQuestion[];
}

export async function generateQuiz(lessonId: string): Promise<AIGeneratedQuiz> {
  return api.request<AIGeneratedQuiz>(`/ai/generate-quiz/${lessonId}`, {
    method: 'POST',
  });
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field
from typing import Optional


class AIGenerateRequest(BaseModel):
    prompt: str = Field(
        ...,
        min_length=10,
        max_length=1000,
        description="Description du thème de la formation",
    )
    model: str = Field(
        default="mistral-small-latest", description="Modèle Mistral à utiliser"
    )


class AILesson(BaseModel):
    title: str
    description: str
    lesson_type: str = "video"
    order: int
    duration_seconds: int


class AIModule(BaseModel):
    title: str
    description: str
    order: int
    lessons: list[AILesson]


class AIGenerateResponse(BaseModel):
    title: str
    slug: str
    short_description: str
    description: str
    level: str
    category: str
    modules: list[AIModule]


class QuizOption(BaseModel):
    label: str
    text: str
    is_correct: bool


class AIQuestion(BaseModel):
    text: str
    order: int
    question_type: str = "single"
    explanation: str = ""
    points: int = 10
    options: list[QuizOption]


class AIGenerateQuizResponse(BaseModel):
    title: str
    passing_score_percent: int = 70
    questions: list[AIQuestion]
